react-events-and-state/src/index.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isClicked: false };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState(prevState => ({ isClicked: true }));
```

Comment:
You can just pass an object to `setState` you don't need to use the callback function, that is for special circumstances.

```JSX
this.setState({ isClicked: true });
```